### PR TITLE
React to aspnet/Razor#196 - Maintain VisualStudioCompatibility Test.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Add new `IsEnum` `TagHelperDescriptor` property to VS compatibility test.

### CODE REVIEW

File: test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs [hunk 1]
```diff
@@ -105,22 +133,30 @@ public void TagHelperDescrictorsAreCompatibleWithPinnedVisualStudioVersion()
...
+                Assert.Equal(attribute.IsEnum, expectedAttribute.IsEnum);
+                Assert.Equal(attribute.IsStringProperty, expectedAttribute.IsStringProperty);
+                Assert.Equal(attribute.Name, expectedAttribute.Name, StringComparer.Ordinal);
+                Assert.Equal(attribute.PropertyName, expectedAttribute.PropertyName, StringComparer.Ordinal);
+                Assert.Equal(attribute.TypeName, expectedAttribute.TypeName, StringComparer.Ordinal);
+
+                var dtAttribute = attribute.DesignTimeDescriptor;
+                var expectedDTAttribute = expectedAttribute.DesignTimeDescriptor;
+                Assert.Equal(dtAttribute.Remarks, expectedDTAttribute.Remarks, StringComparer.Ordinal);
+                Assert.Equal(dtAttribute.Summary, expectedDTAttribute.Summary, StringComparer.Ordinal);
```
Comment (dougbu) [inline test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs]: Can't this be done more simply with a comparator and the `IEnumerable` assertions?

File: test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs [hunk 2]
```diff
@@ -105,22 +133,30 @@ public void TagHelperDescrictorsAreCompatibleWithPinnedVisualStudioVersion()
...
 
-            var attribute = Assert.Single(vsDescriptor.Attributes);
-            var expectedAttribute = Assert.Single(expectedVSDescriptor.Attributes);
-            Assert.Equal(attribute.IsIndexer, expectedAttribute.IsIndexer);
-            Assert.Equal(attribute.IsStringProperty, expectedAttribute.IsStringProperty);
-            Assert.Equal(attribute.Name, expectedAttribute.Name, StringComparer.Ordinal);
-            Assert.Equal(attribute.PropertyName, expectedAttribute.PropertyName, StringComparer.Ordinal);
-            Assert.Equal(attribute.TypeName, expectedAttribute.TypeName, StringComparer.Ordinal);
+            var attributes = vsDescriptor.Attributes.ToArray();
+            var expectedAttributes = expectedVSDescriptor.Attributes.ToArray();
```
Comment (dougbu) [inline test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs]: What ensures these descriptors have the same order?

**Comment by dougbu** [comment]
:watch:

File: test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs (hunk 2 above)
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs]: :+1: Will order by attribute name

File: test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs (hunk 1 above)
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs]: Ehh, I could make a comparator. A bit heavy handed for this though.

**Comment by NTaylorMullen** [comment]
:up: :date:  by ordering attribute descriptors

File: test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs]: Well, this is a pretty ugly block of assertions but :ok:

**Comment by dougbu** [comment]
:shipit: once we see the `RazorTooling` build succeed on the CI with `Razor` bits including your aspnet/Razor#196 change.

**Comment by NTaylorMullen** [comment]
It builds :smile:. Checking this in now.

**Pull request merged**